cpp/ql/lib/semmle/code/cpp/ir/dataflow/DefaultTaintTracking.qll

```diff
@@ -474,6 +474,24 @@ module TaintedWithPath {
     }
   }
 
+  /**
+   * INTERNAL: Do not use.
+   */
+  module Private {
+    /** Gets a predecessor `PathNode` of `pathNode`, if any. */
+    PathNode getAPredecessor(PathNode pathNode) { edges(result, pathNode) }
+
+    /** Gets the element that `pathNode` wraps, if any. */
```

Comment:
Is this different from `result = pathNode.getNode().asExpr() or result = pathNode.getNode().asParameter()`?

Reply:
`pathNode` isn't a `DataFlow::PathNode`. It's a `TaintedWithPath::PathNode` (i.e., our wrapper path nodes that we use for `DefaultTaintTracking`) which doesn't expose any `getNode()` member predicate. This additional layer of path nodes is one of the reasons I don't like `DefaultTaintTracking`.